### DISCUSSION TITLE
How to load an image from a gallery

### DISCUSSION THREAD

**Discussion by Daekgown** [yasirkula/UnityNativeGallery#296] [Q&A]
I took screenshots on Android with Unity, saved them in Gallery, but I don't know how to load them from Gallery.

I worked on using the Native Gallery plug-in to save it in the Gallery

I'd like to ask if you can help me by looking at the script

**Comment by yasirkula** [comment]
You need to use NativeGallery.GetImageFromGallery and let the user pick the image from Gallery. Though be aware that user is free to pick any other photo from Gallery. If you wish, you can keep a separate copy of the image in persistentDataPath, as well. Then you can read it back from there.

**Reply by AdemIssadWytopia** [reply]
What about loading it without user image picker ? If you know the filename, try to load the file on the galery folder ?
Or load all files in the specified folder ? 
Is it possible ?

**Reply by yasirkula** [reply]
@AdemIssadWytopia NativeGallery doesn't have an API to get image paths without user interaction. On Android, sometimes (but not always) the returned path of GetImageFromGallery can point to the actual path of the image. In other occasions and on iOS, the image's source path can't be determined and thus the image will be copied to temporaryCachePath by NativeGallery and the returned path will point to that temporary location. In those occasions, if you want to access the image later without presenting the image picker, you must copy the image to persistentDataPath.
